src/jit/lower.cpp

```diff
@@ -1990,6 +1990,7 @@ GenTree* Lowering::LowerTailCallViaHelper(GenTreeCall* call, GenTree* callTarget
 //
 // Notes:
 //    - Decomposes long comparisons that feed a GT_JTRUE (32 bit specific).
+//    - Decomposes long comparisons that produce a value (X86 specific).
```

Comment:
Decomposition takes care of nodes that produce a long value, comparisons only consume long values.

I think there's a more general question: do we still need decomposition or all it does can now be done during lowering? It's possible that the current approach is influenced by representation issues that precede LIR.